### PR TITLE
Fix #6374: Handle quotation marks in PR title in RunAllTests check

### LINKED ISSUE #6374: [BUG]: Check PR for [RunAllTests] fails if there are quotation marks in the PR title.
### Describe the bug

When I attempted to revert a PR, I ran into:

```php
Run if [[ "Revert "Fix #6258: Deprecated Lollipop"" == *"[RunAllTests]"* ]]; then echo "run_all=true" >> $GITHUB_OUTPUT; fi
  if [[ "Revert "Fix #6258: Deprecated Lollipop"" == *"[RunAllTests]"* ]]; then echo "run_all=true" >> $GITHUB_OUTPUT; fi
shell: /usr/bin/bash -e {0}
env:
  CACHE_DIRECTORY: ~/.bazel_cache
  BAZELISK_GITHUB_TOKEN: ***
/home/runner/work/_temp/7fab2851-4989-443a-bd81-6eb69a05fb38.sh: line 1: unexpected token `newline', conditional binary operator expected
Error: Process completed with exit code 2.
```

GitHub Actions does a textual substitution of `${{ github.event.pull_request.title }}` before the shell sees the script.

The PR title being: Revert "Fix #6258: Deprecated Lollipop"

After substitution, the generated script becomes:

`if [[ "Revert "Fix #6258: Deprecated Lollipop"" == *"[RunAllTests]"* ]]; then ...`

The embedded " characters close the outer quoted string early, so bash sees a malformed `[[ ... ]] ` expression, hence the unexpected token 'newline', conditional binary operator expected.

### Steps To Reproduce

1. Create a PR with a title similar to `Revert "Fix #6258: Deprecated Lollipop"`
2. See that the`Check PR for [RunAllTests]` step fails with the above error.

### Expected Behavior

The PR title should be passed into the script in a safe way that does not break the script.

### Screenshots/Videos

_No response_

### What device/emulator are you using?

_No response_

### Which Android version is your device/emulator running?

_No response_

### Which version of the Oppia Android app are you using?

_No response_

### Additional Context

The two options I can think of to fix this are:
1. Add a regex to strip the quotation marks or any other known dangerous characters
2.   Instead of passing the PR title diectly llike: `if [[ "${{ github.event.pull_request.title }}" == *"[RunAllTests]"* ]]; then echo "run_all=true" >> $GITHUB_OUTPUT; fi`, pass them through env instead like: 

```
- name: Check for RunAllTests label in title
  env:
    PR_TITLE: ${{ github.event.pull_request.title }}
  run: |
    if [[ "$PR_TITLE" == *"[RunAllTests]"* ]]; then
      echo "run_all=true" >> "$GITHUB_OUTPUT"
    fi

```

### PR DESCRIPTION
**Pull request by Kishan8548**
## Explanation
Fixes #6374

In `.github/workflows/code_coverage.yml` and `.github/workflows/unit_tests.yml`, the `Check PR for [RunAllTests]` step previously interpolated `${{ github.event.pull_request.title }}` directly into the inline bash script body.

When a PR title contains quotation marks (such as revert PRs e.g. `Revert "Fix #6258: Deprecated Lollipop"`), GitHub Actions' textual substitution breaks outer quotes in the bash script and causes a syntax error:
```
line 1: unexpected token `newline', conditional binary operator expected
```

This PR resolves the issue by passing `${{ github.event.pull_request.title }}` into the step as an environment variable (`PR_TITLE: ${{ github.event.pull_request.title }}`) and referencing `"$PR_TITLE"` inside bash. This ensures the title is passed safely as a process-level environment variable without shell syntax breakage or script injection risk.

### Verification
- Reproduced the syntax error locally using titles with quotation marks under direct interpolation.
- Verified that passing through `PR_TITLE` avoids syntax errors on titles with quotation marks and accurately matches `[RunAllTests]`.
- Validated YAML syntax for both workflow files.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The explanation section above starts with "Fixes #bugnum: " (If this PR fixes part of an issue, use instead: "Fixes part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## Disclosure of LLM Usage

- **Did you use AI/LLMs when working on this PR?** Yes
- **If yes, describe the extent AI was used:** Formulating edge cases to test in bash and formatting the PR description.

### CODE REVIEW

File: .github/workflows/code_coverage.yml [hunk 1]
```diff
@@ -92,8 +92,10 @@ jobs:
       - name: Check PR for [RunAllTests]
         id: check_pr
         if: ${{ github.event_name == 'pull_request' }}
+        env:
+          PR_TITLE: ${{ github.event.pull_request.title }}
         run: |
-          if [[ "${{ github.event.pull_request.title }}" == *"[RunAllTests]"* ]]; then echo "run_all=true" >> $GITHUB_OUTPUT; fi
+          if [[ "$PR_TITLE" == *"[RunAllTests]"* ]]; then echo "run_all=true" >> $GITHUB_OUTPUT; fi
```
[the remark this answers is not in the document]
Comment (Kishan8548) [reply .github/workflows/code_coverage.yml:98]: this was already duplicated before, keeping this pr scoped only to fixing the title issue

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -82,8 +82,10 @@ jobs:
       - name: Check PR for [RunAllTests]
         id: check_pr
         if: ${{ github.event_name == 'pull_request' }}
+        env:
+          PR_TITLE: ${{ github.event.pull_request.title }}
         run: |
-          if [[ "${{ github.event.pull_request.title }}" == *"[RunAllTests]"* ]]; then echo "run_all=true" >> $GITHUB_OUTPUT; fi
+          if [[ "$PR_TITLE" == *"[RunAllTests]"* ]]; then echo "run_all=true" >> $GITHUB_OUTPUT; fi
```
[the remark this answers is not in the document]
Comment (Kishan8548) [reply .github/workflows/unit_tests.yml:88]: this was already written like this in the workflow and works fine on runners, keeping changes minimal to the issue

**Comment by adhiamboperes** [comment]
Re-triggering CI in order to try to trip the same path as the original bug.

**Review by adhiamboperes — APPROVED** [review]
Verified that the bug is fixed. LGTM, thanks!
